dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -197,13 +282,11 @@ class FunctionsStringSearch : public IFunction
         auto block = result_block;
         if constexpr (name == std::string_view(NameIlike3Args::name))
         {
-            if (!collator->isCI())
-            {
-                block.getByPosition(arguments[0]).column = (*std::move(result_block.getByPosition(arguments[0]).column)).mutate();
-                block.getByPosition(arguments[1]).column = (*std::move(result_block.getByPosition(arguments[1]).column)).mutate();
+            block.getByPosition(arguments[0]).column = (*std::move(result_block.getByPosition(arguments[0]).column)).mutate();
+            block.getByPosition(arguments[1]).column = (*std::move(result_block.getByPosition(arguments[1]).column)).mutate();
 
-                IlikeHelper::lowerAlphaASCII(block, arguments);
-            }
+            IlikeLowerHelper::lowerAlphaASCII(block, arguments);
+            IlikeLowerHelper::convertCollatorToBin(collator);
```

Comment:
Why add this convert?

Reply:
> Why add this convert?

The following is the reasons:
- ci behaviour of ci collator is not expected by us, it will do ci for non-ascii chars
- bin collator is the fastest collator